src/vulkan/vulkanutils.cpp

```diff
@@ -3,9 +3,140 @@
 #include <iomanip>
 #include <sstream>
 
-// TODO: fix namespaces
+#if defined(__APPLE__) || defined(__linux)
+#include <dlfcn.h>
+#endif
+
 namespace dw::vulkan {
 
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+VulkanRTLPtr GetRuntimeLibs()
+{
+    VulkanRTLPtr vulkanLibrary;
+
+#if defined (_WIN32)
+    vulkanLibrary = LoadLibrary("vulkan-1.dll");
+#elif defined (__linux )
+    vulkanLibrary = dlopen("libvulkan.so.1", RTLD_NOW | RTLD_LOCAL);
+#elif defined (__APPLE__ )
+    vulkanLibrary = dlopen("libvulkan.1.dylib", RTLD_NOW | RTLD_LOCAL);
+#else
+    std::cerr << "The DireWolf renderer is not yet setup for Vulkan on this platform. Supported operating systems are Linux, Windows and Mac" << std::endl;
+    vulkanLibrary = nullptr;
+#endif
+
+    if (!vulkanLibrary) {
+        std::cerr << "Could not connect with a Vulkan Runtime library.\n";
+    }
+
+#if defined (DW_VERBOSE_LOG_VK)
+    std::cout << "\tSuccessfully connected with a Vulkan Runtime library.\n";
+#endif
+    return vulkanLibrary;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+bool InitProcAddress(VulkanRTLPtr vulkanRTL)
+{
+// TODO: move this #VK_NO_PROTOTYPES part elsewhere?
+#if defined (DW_VERBOSE_LOG_VK) && defined (VK_NO_PROTOTYPES)
+    std::cout << "\tVK_NO_PROTOTYPES is defined\n";
+#elif defined (DW_VERBOSE_LOG_VK) && !defined (VK_NO_PROTOTYPES)
+    std::cout << "\tWARNING: VK_NO_PROTOTYPES is NOT defined. This may become a potential performance issue.\n";
+#endif
+
+#if defined (_WIN32)
+#define LoadFunction GetProcAddress
```

Comment:
Half indent or what did we decide

Reply:
Yeah, I'm all for readability so I think I'd prefer some type of indentation at least.. half/full?